roofit/histfactory/src/ConfigParser.cxx

```diff
@@ -274,115 +272,98 @@ HistFactory::Measurement ConfigParser::CreateMeasurementFromDriverNode( TXMLNode
   measurement.SetBinHigh( 1 );
   measurement.SetExportOnly( false );
 
-  cxcoutIHF << "Creating new measurement: " << std::endl;
+  cxcoutIHF << "Creating new measurement: \n";
 
   // First, get the attributes of the node
   TListIter attribIt = node->GetAttributes();
-  TXMLAttr* curAttr = 0;
-  while( ( curAttr = dynamic_cast< TXMLAttr* >( attribIt() ) ) != 0 ) {
-
-    if( curAttr->GetName() == TString( "" ) ) {
-      cxcoutEHF << "Found XML attribute in Measurement with no name"  << std::endl;
+  TXMLAttr* curAttr = nullptr;
+  while ( (/**/curAttr = dynamic_cast<TXMLAttr*>(attribIt())/**/) != nullptr ) {
+    const char * curAttrName = curAttr->GetName();  // curAttr is guaranteed non-null
+    const char * curAttrValue = curAttr->GetValue();
+    // we need to explicitly check for nullptr returned from GetName()
+    // as it will never equal an empty TString()
+    if ( curAttrName == nullptr || curAttrName == TString("") ) {
+      cxcoutEHF << "Found XML attribute in Measurement with no name.\n";
       // ADD Output Here
       throw hf_exc();
-    }
-    else if( curAttr->GetName() == TString( "Name" ) ) {
-      //rowTitle=curAttr->GetValue();
-      measurement.SetName(  curAttr->GetValue() );
-      //measurement.OutputFileName = outputFileNamePrefix+"_"+rowTitle+".root";
-    }
-    else if( curAttr->GetName() == TString( "Lumi" ) ) {
-      measurement.SetLumi( atof(curAttr->GetValue()) );
-    }
-    else if( curAttr->GetName() == TString( "LumiRelErr" ) ) {
-      measurement.SetLumiRelErr( atof(curAttr->GetValue()) );
-    }
-    else if( curAttr->GetName() == TString( "BinLow" ) ) {
-      measurement.SetBinLow( atoi(curAttr->GetValue()) );
-    }
-    else if( curAttr->GetName() == TString( "BinHigh" ) ) {
-      measurement.SetBinHigh( atoi(curAttr->GetValue()) );
-    }
-    else if( curAttr->GetName() == TString( "Mode" ) ) {
-      cout <<"\n INFO: Mode attribute is deprecated, will ignore\n"<<endl;
-    }
-    else if( curAttr->GetName() == TString( "ExportOnly" ) ) {
-      measurement.SetExportOnly( CheckTrueFalse(curAttr->GetValue(),"Measurement") );
-    }
-
-    else {
-      cxcoutEHF << "Found unknown XML attribute in Measurement: " << curAttr->GetName()
-		<< std::endl;
+    } else if ( curAttrName == TString("Name") ) {
+      // rowTitle=curAttr->GetValue();
+      measurement.SetName(curAttrValue);
+      // measurement.OutputFileName = outputFileNamePrefix + "_" + rowTitle + ".root";
+    } else if ( curAttrName == TString("Lumi") ) {
+      measurement.SetLumi(atof(curAttrValue));
+    } else if ( curAttrName == TString("LumiRelErr") ) {
+      measurement.SetLumiRelErr(atof(curAttrValue));
+    } else if ( curAttrName == TString("BinLow") ) {
+      measurement.SetBinLow(atoi(curAttrValue));
+    } else if ( curAttrName == TString("BinHigh") ) {
```

Comment:
... and making those
```c++
else if ( curAttrName == "LumiRelErr" )
```
?

There's no need to create new TStrings for every `if`.

Reply:
True, wasn't sure how you preferred this internally. I'll update.